keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -248,26 +254,38 @@ implementation. What is the desired outcome and how do we measure success?.
 The "Design Details" section below is for the real
 nitty-gritty.
 -->
-We propose using an opt-in API mechanism (such as content-type header or query
-param) to indicate to the server that it should fail when the kubernetes object
-in the request body supplied to POST, PUT, and PATCH requests contains
-extra/unknown fields.
 
-Clients such as kubectl will continue to use the `--validate=true` flag as
-before, but instead of triggering validation on the client-side, it will
-instruct the server to validate for unknown fields server-side.
+We propose adding server-side schema validation, enabling the API server
+to validate that objects received in the request body in POST, PUT, and
+PATCH requests contain no unknown or duplicate fields.
 
-This change will be made in at least two steps, one where we introduce the
-server side validation and a second where we modify kubectl to use the
-server-side validation (and mark the existing client-side validation as
-deprecated).
+We introduce a query parameter that enables the user/client to indicate
+that server-side validation should either not be performed at all, provide
+warnings for the presence of unknown/duplicate fields, or error out in the
+presence of unknown/duplicate fields.
+
+For beta, we propose modifying the kubectl `--validate` flag to enable users
+to choose between client-side validation or server-side
+validation (either strict or warning)
 
 ### Notes/Constraints/Caveats (Optional)
 
 #### Future Work
-After server side unknown field validation is implemented we can begin work to
-deprecate and remove client side validate and have kubectl use server side
-validation instead. A separate KEP will be published for that.
+
+Upon successfully providing kubectl access to server-side validation via the
+`--validate` flag, an open question will remain as to the future of client
+side validation.
+
+While we would like kubectl to use server-side validation by default once it
+is GA, the question remains whether client-side validation be optionally
+available to kubectl users or completely deprecated.
+
+Maintaining client-side validation indefinitely provides the benefit of
+allowing users to validate their objects without needing access to a server.
+On the other hand, the downside of continuing to maintain client-side
+validation is that it uses openapiv2 which will be less actively supported
+as openapiv3 becomes standard and will cause client-side validation to
+continue to diverge from server-side validation.
```

Comment:
Are there any mechanisms available to address this (if not, what's the most obvious obstacle)?

Reply:
I've added a note about separate "out-of-tree" solutions for CS validation as pre-req for deprecating the existing CS validation, noting that OpenAPI v3 support for kubeval is one such solution that needs further investigation.

Does this answer your question?

CC @apelisse because I think you have better context on what specificially will break CS validation as openapiv3 progresses in adoption